src/tools/kata-ctl/src/check.rs

```diff
@@ -90,6 +92,16 @@ pub fn run_network_checks() -> Result<()> {
 }
 
 pub fn check_version() -> Result<()> {
+    let response = reqwest::blocking::Client::new()
+        .get("https://api.github.com/repos/kata-containers/kata-containers/releases/latest")
+        .header(CONTENT_TYPE, "application/json")
+        .header(USER_AGENT, "kata")
+        .send()
+        .expect("Failed to retrieve latest kata version.");
```

Comment:
This is neat! Could you restructure the code slightly though to make this unit-testable. I'd do something like

```rust
const KATA_GITHUB_URL: &str = "https://api.github.com/repos/kata-containers/kata-containers/releases/latest";

/// Returns the latest version of Kata as a semver string.
fn get_kata_version_by_url(url: &str) -> Result<String> {
    // ...
}

pub fn check_version() -> Result<()> {
    let version = get_kata_version_by_url(KATA_GITHUB_URL)?;

    println!("Version: {}", ver);

    Ok(())
}
```

That way, we can unit-test `check_version_by_url()`:

- Specify empty/invalid/garbage URL and assert that an `Err` value is returned.
- Specify `KATA_GITHUB_URL` and ensure that a non-empty and valid semver string is returned.

A few other suggestions:

- Could you change the `.expect()` to `.context("Failed to retrieve latest kata version") (you'll need to add `use anyhow::Context`) so that the error is returned by that function.
- Please could you validate that the string returned by GitHub is valid semver.

Aside: We'll have to tweak this code slightly when we add `--include-all-releases`, but that doesn't need to be done on this PR.